test/specs/rules/validate-comment-position.js

```diff
@@ -0,0 +1,186 @@
+var Checker = require('../../../lib/checker');
+var assert = require('assert');
+
+describe('rules/validate-comment-position', function() {
+    var checker;
+
+    beforeEach(function() {
+        checker = new Checker();
+        checker.registerDefaultRules();
+    });
+
+    it('accepts valid mode', function() {
+        var validModes = [
+            'above',
+            'beside'
+        ];
+
+        validModes.forEach(function(mode) {
+            assert.doesNotThrow(function() {
+                checker.configure({ validateCommentPosition: { mode: mode }});
+            });
+        });
+    });
+
+    it('rejects invalid mode', function() {
+        var invalidModes = [
+            'beneath',
+            'under',
+            'perpendicular',
+            true
+        ];
+
+        invalidModes.forEach(function(mode) {
+            assert.throws(function() {
+                checker.configure({ validateCommentPosition: { mode: mode }});
+            }, assert.AssertionError);
+        });
+    });
+
+    it('accepts valid exceptions', function() {
+        assert.doesNotThrow(function() {
+            checker.configure({ validateCommentPosition: { mode: 'above', allExcept: ['pragma', 'linter'] }});
+        });
+    });
+
+    it('rejects invalid exceptions', function() {
+        var invalidExceptions = [
+            [1, 2, 3],
+            ['linter', 1, true]
+        ];
+
+        invalidExceptions.forEach(function(exception) {
+            assert.throws(function() {
+                checker.configure({ validateCommentPosition: { mode: 'above', allExcept: exception }});
+            }, assert.AssertionError);
+        });
+    });
+
+    describe('mode value "above"', function() {
+        beforeEach(function() {
+            checker.configure({ validateCommentPosition: { mode: 'above' }});
+        });
+
+        it('should report unexpected position for 1 + 1; // invalid comment', function() {
```

Comment:
Non-blocker: we typically don't put code in the descriptions. A general description of the code would suffice. Like `it('should report on a comment beside a statement', ...`